### PR TITLE
Defendpoint autogen dox polish

### PR DESCRIPTION
**Pull request by escherize**
- doc generation should behave the same way for malli or plumatic schema.
- fixes wording a bit in umd/describe (esspecially for empty maps)
- switched out 2 routes, and regenerated dox so you can review 1 simple and 1 complex route change.
- switched over to new defendpoint macro for the metabase.api.action namespace + verified doc generation

### CODE REVIEW

File: docs/api/setting.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Fetch a single `Setting`.
 
 ### PARAMS:
 
-*  **`key`** value must be a non-blank string.
+*  **`key`** string with length longer than 1
```
Comment (escherize) [inline docs/api/setting.md:23]: Caused by: https://github.com/metabase/metabase/blob/bcm-defendpoint-autogen-dox-polish/src/metabase/api/setting.clj#L44

`{key su/NonBlankString}` -> `{key [:string {:min 1}]}`

Comment (escherize) [inline docs/api/setting.md:23]: Caused by: https://github.com/metabase/metabase/blob/bcm-defendpoint-autogen-dox-polish/src/metabase/api/setting.clj#L44

`{key su/NonBlankString}` -> `{key [:string {:min 1}]}`

File: docs/api/action.md [hunk 2]
```diff
@@ -57,24 +57,17 @@ Create a new HTTP action.
...
-  method : value must be one of: `DELETE`, `GET`, `PATCH`, `POST`, `PUT`.
-  url : value must be a string.
-)
+*  **`template`** a nullable a map where {:method -> <an enum of GET, POST, PUT, DELETE, PATCH>, :url -> <string with length longer than 1>, :body (optional) -> <a nullable string>, :headers (optional) -> <a nullable string>, :parameters (optional) -> <a nullable sequence of a map >, :parameter_mappings (optional) -> <a nullable a map >} with no other keys
 
-*  **`response_handle`** value may be nil, or if non-nil, must be a valid json-query
+*  **`response_handle`** a nullable string, and must be a valid json-query
 
-*  **`error_handle`** value may be nil, or if non-nil, must be a valid json-query
+*  **`error_handle`** a nullable string, and must be a valid json-query
```
Comment (escherize) [inline docs/api/action.md:71]: caused by: 

https://github.com/metabase/metabase/blob/bcm-defendpoint-autogen-dox-polish/src/metabase/api/action.clj#L86-L112

File: src/metabase/util/malli/describe.clj [hunk 3]
```diff
@@ -86,16 +86,18 @@
...
+  (str "map " (titled schema) "from " (diamond (first children)) " to " (diamond (second children)) (min-max-suffix schema)))
 
-(defmethod accept 'vector? [_ schema children _] (str "vector" (titled schema) (min-max-suffix schema) " of " (first children)))
-(defmethod accept :vector [_ schema children _] (str "vector" (titled schema) (min-max-suffix schema) " of " (first children)))
+(defn- of-clause [children] (when children (str " of " (first children))))
 
-(defmethod accept 'sequential? [_ schema children _] (str "sequence" (titled schema) (min-max-suffix schema) " of " (first children)))
-(defmethod accept :sequential [_ schema children _] (str "sequence" (titled schema) (min-max-suffix schema) " of " (first children)))
+(defmethod accept 'vector? [_ schema children _] (str "vector" (titled schema) (min-max-suffix schema) (of-clause children)))
+(defmethod accept :vector [_ schema children _] (str "vector" (titled schema) (min-max-suffix schema) (of-clause children)))
```
Comment (escherize) [inline src/metabase/util/malli/describe.clj:95]: the diff is not good here. we still have vector, sequential, and set...

**Review by camsaul — APPROVED** [review]

File: src/metabase/api/action.clj [hunk 4]
```diff
@@ -84,14 +84,30 @@
...
+         :description "must be a valid json-query"}
+    #(http-action/apply-json-query {} %)]])
+
+(def ^:private http-action-template
+  [:map {:closed true}
+   [:method                              [:enum "GET" "POST" "PUT" "DELETE" "PATCH"]]
+   [:url                                 [string? {:min 1}]]
+   [:body               {:optional true} [:maybe string?]]
+   [:headers            {:optional true} [:maybe string?]]
+   [:parameters         {:optional true} [:maybe [:sequential map?]]]
```
Comment (qnkhuat) [inline src/metabase/api/action.clj:100]: a question: what is the different if we use `:*` instead of `:sequential` here?

**Review by qnkhuat — APPROVED** [review]
awesome

File: src/metabase/api/action.clj (hunk 4 above)
Comment (escherize) [reply src/metabase/api/action.clj:100]: I think they're the same, but I think of `:*` as being part of the sequence regex schemas, so I wouldn't reach for it unless you're doing something that requires it. [regex schema ex.](https://malli.io/?value=%5B%3Anames%20%5B%22a%22%20%22b%22%5D%20%3Anums%20%5B1%202%203%5D%5D&schema=%5B%3Acat%20%5B%3A%3D%20%3Anames%5D%20%5B%3Aschema%20%5B%3A*%20string%3F%5D%5D%20%5B%3A%3D%20%3Anums%5D%20%5B%3Aschema%20%5B%3A*%20number%3F%5D%5D%5D)

File: docs/api/setting.md (hunk 1 above)
Comment (noahmoss) [reply docs/api/setting.md:23]: Is this accurate? I think of "longer than 1" meaning 2 or more.

**Pull request merged**